Apps/phgsgmail/gsgmail_connector.py

```diff
@@ -243,6 +248,99 @@ def _map_email_details(self, input_email):
+        plain_bodies = []
+        html_bodies = []
+        container_id = self.get_container_id()
+
+        email_details['email_headers'] = []
+        for part in msg.walk():
+            type = part.get_content_type()
+            headers = self._get_email_headers_from_part(part)
+            # split out important headers (for output table rendering)
+            if headers.get('to'):
+                email_details['to'] = headers.get('to')
+
+            if headers.get('from'):
+                email_details['from'] = headers.get('from')
+
+            if headers.get('subject'):
+                email_details['subject'] = headers.get('subject')
+
+            disp = str(part.get('Content-Disposition'))
+            file_name = part.get_filename()
+            # look for plain text parts, but skip attachments
+            if type == 'text/plain' and 'attachment' not in disp:
+                charset = part.get_content_charset() or 'utf8'
+                # decode the base64 unicode bytestring into plain text
+                plain_body = part.get_payload(decode=True).decode(encoding=charset, errors="ignore")
+                # Add to list of plan text bodies
+                plain_bodies.append(plain_body)
+            if type == 'text/html' and 'attachment' not in disp:
+                charset = part.get_content_charset() or 'utf8'
+                # decode the base64 unicode bytestring into plain text
+                html_body = part.get_payload(decode=True).decode(encoding=charset, errors="ignore")
+                # Add to list of html bodies
+                html_bodies.append(html_body)
+            elif file_name and extract_attachments:
+                attach_resp = None
+                try:
+                    # Create vault item with attachment payload
+                    attach_resp = Vault.create_attachment(part.get_payload(decode=True), container_id=container_id, file_name=file_name)
+                except Exception as e:
+                    self.debug_print('Unable to add attachment: {} Error: {}').format(str(file_name), str(e))
+                if attach_resp.get('succeeded'):
+                    # Create vault artifact
+                    artifact = {
+                        'name': 'Email Attachment Artifact',
+                        'container_id': container_id,
+                        'cef': {
+                            'vaultId': attach_resp[phantom.APP_JSON_HASH],
+                            'fileHash': attach_resp[phantom.APP_JSON_HASH],
+                            'file_hash': attach_resp[phantom.APP_JSON_HASH],
+                            'fileName': file_name
+                        },
+                        'run_automation': False,
+                        'source_data_identifier': None
+                    }
+                    ret_val, _, _ = self.save_artifact(artifact)
+                    if phantom.is_fail(ret_val):
+                        return action_result.get_status()
```

Comment:
suggestion: you can also return `phantom.APP_ERROR` I think. This would help remove the param `action_result` here.

Reply:
So I think I understand. Usually functions take action_result as the param and you use action_result.get_status() to propagate the error. In this case, the function just returns a message. I changed that in the recent commit